src/classes/AuthenticationSession.ts

```diff
@@ -13,10 +13,20 @@ enum FusionAuthStatusCode {
 }
 
 export class AuthenticationSession {
+  private static readonly sftpFusionAuthAppId = process.env.FUSION_AUTH_SFTP_APP_ID ?? '';
```

Comment:
We won't need them as instance properties with https://www.npmjs.com/package/require-env-variable

Reply:
@Fenn-CS I hear you, but I think it would be more explicit / better practice to have them as instance properties and require them as part of the construction of an `AuthenticationSession`.

Reply:
@slifty all we need to with the `require-env-variable` package is to import that variables and it handles the rest including failure and exit in case of env var absence.

I mean

```js
const {
  FUSION_AUTH_SFTP_APP_ID,
  FUSION_AUTH_SFTP_CLIENT_ID,
  FUSION_AUTH_SFTP_CLIENT_SECRET,
} = requireEnv(
  'FUSION_AUTH_SFTP_APP_ID',
  'FUSION_AUTH_SFTP_CLIENT_ID',
  'FUSION_AUTH_SFTP_CLIENT_SECRET',
);
```

Since we are not using any of these `env` vars upper in the stack, those would become unused imports, unless we want to force use them without need to annotate the import so that it is ignored by the linter. 

I think it's still okay for the system to exit at the point where it needs something and it does not exist it must not be at the very top.

Reply:
I hear you from an implementation perspective, but I want to be sure what I'm saying is clear because I'm talking about code design!

This class requires these values to function, and having these values as part of the construction signature is *more explicit* and signals that requirement to any developer that might use the class.  This does a few things:

1. If there was ever a need to use this class to authenticate with multiple fusion app IDs, we could have different instances to handle those different IDs.

2. It decouples the validation of these values from this class (to your point, validation right now is handled by the requireEnv library, but imagine a situation where validation was more involved)

3. It allows us to recognize a failure in configuration instantly on deployment -- which is very important in a production environment.

What I'm proposing is that you require these things at `server.ts` and then pass them from class to class -- the Ssh handler class would need to know what the App ID, Client ID, and Client Secret for its SFTP subcomponent are supposed to be.

As of now there is no way, for example, (without inspecting or searching the entire code base) for a developer to realize that these environment variables are required in order for the SshSessionHandler to function!  That's not ideal and contributes to tech debt over time.

NOTE: if you and / or @jasonaowen think that explicitness isn't worth the trouble in this case then I totally understand!  I just wanted to make sure I was clear in explaining that what I was suggesting wasn't just to address implementation challenges.

Reply:
Thanks @slifty that's much clearer and can be done without much hassle. I would add this final bit of context to know your thoughts, then go ahead. So, what `requireEnv` actually does is that, you won't be able to build the source or `run dev` at all if the environment variables are not present. Given that, and that, the class that directly depends on these `AuthenticationSession` now requires these as instance values passed in the constructor, do you still think it's important for us to pass these values about 2 - 3 classes down just to get in to the `AuthenticationSession` class?

Reply:
You know what? It's one class down, I would just do that!

Reply:
That's good context (that run dev would immediately break) -- and that does address the biggest concern from a practical perspective.

The code design question is a little bit more of a philosophical question and I am completely open to the idea that what I'm suggesting is not worth the tradeoff in terms of code complexity.

All this is to say: at this point it sounds like you totally grok what I was trying to explain, and I defer to you / the group on what you think is best!  Thank you for coming to my TED talk 😆

Reply:
It's done @slifty so please kindly review, I won't mark this as resolved because we had this conversation I would leave that to you.